File: PlaceHolderView/PlaceHolderView/ViewController.m
//
//  ViewController.m
//  PlaceHolderView
//
//  Created by Wang HaiLong on 13/2/15.
//  Copyright (c) 2015 Hai Long. All rights reserved.
//

#import "ViewController.h"
#import "PlaceholderView.h"

@implementation ViewController{

    PlaceholderView *placeholderView;

}

- (void)viewDidLoad {
    [super viewDidLoad];

    placeholderView = [[PlaceholderView alloc]initWithFrame:self.webView.frame];

    [placeholderView setPlaceholderImage:[UIImage imageNamed:@"info"] withImageSize:CGSizeMake(60, 60) andMessage:@"Sorry, Can not connect to server, please check network configuration"];
    [placeholderView addPlaceholderTapTarget:self andAction:@selector(reloadWeb)];

    [self.view addSubview:placeholderView];

    [self.webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://www.google.com.sg"]]];
}

-(void)reloadWeb{

    [self.webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:[NSString stringWithFormat:@"https://www.google.com.sg/#newwindow=1&q=%@",self.searchBar.text]]]];
}

- (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar{
    [self.webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:[NSString stringWithFormat:@"https://www.google.com.sg/#newwindow=1&q=%@",searchBar.text]]]];
    [searchBar resignFirstResponder];
}

- (void)webViewDidStartLoad:(UIWebView *)webView{
    placeholderView.isInProcess = YES;
}

- (void)webViewDidFinishLoad:(UIWebView *)webView{
    placeholderView.isInProcess = NO;
    placeholderView.hidden = YES;
}

- (void)webView:(UIWebView *)webView didFailLoadWithError:(NSError *)error{
    placeholderView.isInProcess = NO;
    placeholderView.hidden = NO;
}



@end

File: HLPlaceholderViewDemo/ViewController.swift
//
//  ViewController.swift
//  HLPlaceholderViewDemo
//
//  Created by PandaApe on 7/12/15.
//  Copyright (c) 2015 PandaApe. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UIWebViewDelegate {
    
    @IBOutlet weak var placeholderView: HLPlaceholderView!
    @IBOutlet weak var webView: UIWebView!
    
    var sampleAddress = "https://github.com/pandaApe/HLPlaceHolderView"
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        placeholderView.imgOfIconImgView = UIImage(named: "info")
        placeholderView.addPlaceholderTapTarget(self, andAction: #selector(ViewController.loadData))
        placeholderView.sizeOfIconImageView = CGSizeMake(60, 60)
        
        webView.delegate = self;
        webviewLoadWithURLStr(sampleAddress)
    }
    
    func loadData(){
        webView.reload()
    }
    
    //UIWebview Delegate Method
    func webViewDidStartLoad(webView: UIWebView) {
        placeholderView.startAnimating()
    }
    
    func webView(webView: UIWebView, didFailLoadWithError error: NSError?) {
        placeholderView.stopAnimating()
        placeholderView.msgText = "\(error?.domain) \nFailed!" ;
    }
    
    func webViewDidFinishLoad(webView: UIWebView) {
        placeholderView.stopAnimating()
        placeholderView.hidden = true
    }
    
    @IBAction func switchValueChanged(sender: UISwitch) {
        sender.tag *= -1
        var addressUrl:String
        if sender.tag > 0 {
            addressUrl = sampleAddress
        }else{
            addressUrl = "-"+sampleAddress
        }
        
        webviewLoadWithURLStr(addressUrl)
        
    }
    
    func webviewLoadWithURLStr(address:String){
        webView.loadRequest(NSURLRequest(URL: NSURL(string: address)!))
    }
    
    
    
}

